checks/evaluation/fuzzing/fuzzing.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2021 OpenSSF Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package evaluation
+
+import (
+	"embed"
+	"fmt"
+
+	"github.com/ossf/scorecard/v4/checker"
+	sce "github.com/ossf/scorecard/v4/errors"
+	"github.com/ossf/scorecard/v4/finding"
+)
+
+//go:embed *.yml
+var rules embed.FS
+
+var (
+	fuzzingWithClustterFuzzLite = "FuzzingWithClusterFuzzLite"
+	fuzzingWithOneFuzz          = "FuzzingWithOneFuzz"
+	fuzzingWithOSSFuzz          = "FuzzingWithOSSFuzz"
+	fuzzingWithGoNative         = "FuzzingWithGoNative"
+)
+
+// Fuzzing applies the score policy for the Fuzzing check.
+func Fuzzing(name string, dl checker.DetailLogger,
+	r *checker.FuzzingData,
+) checker.CheckResult {
+	if r == nil {
+		e := sce.WithMessage(sce.ErrScorecardInternal, "empty raw data")
+		return checker.CreateRuntimeErrorResult(name, e)
+	}
+
+	// Keep track of reported results.
+	reportedRuleResults := make(map[string]bool)
+
+	if len(r.Fuzzers) == 0 {
+		// Report findings for all rules.
+		if err := logDefaultFindings(dl, reportedRuleResults); err != nil {
+			return checker.CreateRuntimeErrorResult(name, err)
+		}
+		return checker.CreateMinScoreResult(name, "project is not fuzzed")
+	}
+	fuzzers := []string{}
+	for i := range r.Fuzzers {
+		fuzzer := r.Fuzzers[i]
+		// NOTE: files are not populated for all fuzzers yet.
+		// To simplify the code, we currently do not report the locations.
+		switch fuzzer.Name {
```

Comment:
there's a lot of similarity here that could probably be extracted out if we had a map from fuzzer.Name -> rule name.

The switch statement could be replaced with a map. Something like



```
m := map[string]string{
    "GoNativeFuzzer": fuzzingWithGoNative,
    // and so on
    // ...
}

fuzzRule, ok := m[fuzzer.Name]
if !ok {
 // unsupported rule
}
if err := checker.LogFinding(rules, fuzzRule,
  "project fuzzed with " + fuzzer.Name,
  nil, finding.OutcomePositive, dl); err != nil {
  return checker.CreateRuntimeErrorResult(name, err)
}
reportedRuleResults[fuzzRule] = true
```

WDUT?